DragonSlayer.cpp

```diff
@@ -1,19 +1,31 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer (std::string name_, int hp_, int armor_):
+Character(hp_, armor_, 80),
+name { name_ }
+{
+    helpfulItems = makeHelpfulItems(2);
+    defensiveItems = makeDefensiveItems(0);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        //assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //
+        attackItem->use(this); //call the AttackItems use function on the DragonSlayer
```

Comment:
items are usable ONCE.  take a look at what happens when you use a helpful item or defensive item in Character.cpp.
- [ ] revise.